src/common/models/Cookie.class.js

```diff
@@ -766,6 +756,32 @@ class Cookie {
     return `${h}:${m}`;
   }
 
+  parseVoyageObj(type, voyageObj) {
+    let voyage = {};
+    if (!voyageObj[`${type}Date`]) {
+      voyage[`${type}Date`] = this.generateDate(
+        voyageObj[`${type}Day`],
+        voyageObj[`${type}Month`],
+        voyageObj[`${type}Year`]
+      );
+
+      voyage[`${type}Time`] = this.generateTime(voyageObj[`${type}Hour`],
+        voyageObj[`${type}Minute`]);
+    } else {
+      // Set voyage from API response, so dates and times are already built
+      voyage[`${type}Date`] = voyageObj[`${type}Date`];
+      voyage[`${type}Time`] = voyageObj[`${type}Time`];
+    }
+    voyage[`${type}Port`] = voyageObj[`${type}Port`];
+    voyage[`${type}Long`] = trimToDecimalPlaces(voyageObj[`${type}Long`], 6);
+    voyage[`${type}Lat`] = trimToDecimalPlaces(voyageObj[`${type}Lat`], 6);
+    
+    const defaultPortChoice = (voyageObj[`${type}Lat`] || voyageObj[`${type}Long`]) ? 'No' : 'Yes';
```

Comment:
Consider splitting the logic here to improve readability, e.g. 

```
 const hasCoordinates = voyageObj[`${type}Lat`] || voyageObj[`${type}Long`];
    const defaultPortChoice = hasCoordinates ? 'No' : 'Yes';
```

Reply:
I don't think it matters enough to merit changing.
* One can argue that it is being over split at this point, making it harder to follow.
* In truth, it is kinda splitting hairs.
* If I am wrong let me know.